doc/designs/quic-design/error-handling.md

```diff
@@ -0,0 +1,88 @@
+Error handling in QUIC code
+===========================
+
+Current situation with TLS
+--------------------------
+
+The errors are put on the error stack (rather a queue but error stack is
+used throughout the code base) during the libssl API calls. In most
+(if not all) cases they should appear there only if the API call returns an
+error return value. The `SSL_get_error()` call depends on the stack being
+clean before the API call to be properly able to determine if the API
+call caused a library or system (I/O) error.
```

Comment:
Side note: Although that is the way `SSL_get_error()` currently works we could actually easily change this by querying `ossl_statem_in_error()`. That would be a change of behaviour though.

Reply:
Would such change of behavior be acceptable in 3.2? IMO it would be nice to unify its behavior between TLS/DTLS and QUIC.

Reply:
I'm unsure. Probably that's a question for OTC.

Reply:
I would certainly support changes which reduce the dependence of SSL_get_error on the ERR state.